cli/src/asset_identifier.rs

```diff
@@ -0,0 +1,66 @@
+use error::InvalidInputError;
+use stellar_client::resources::AssetIdentifier;
+
+pub fn parse_identifier_from_str(s: &str) -> Result<AssetIdentifier, InvalidInputError> {
+    let asset_parts: Vec<&str> = s.split("-").collect();
+    if asset_parts[0] == "xlm" {
+        return Ok(AssetIdentifier::Native);
+    }
+    if asset_parts.len() != 2 {
```

Comment:
You may try using pattern matching for this https://play.rust-lang.org/?gist=85d7266724a99db847d6741730ddebe6&version=stable&mode=debug

Reply:
*reacts with 🔥 emoji*